CDCgov/prime-reportstream
# 7277 Codified Error Messages

1. Updated the ActionMessages.kt -> InvalidHL7Message() method to take an optional ‘ErrorCode’ string
2. Created an enum ‘ErrorCode’ to allow for consistent nomenclature and containment of the error codes to be passed into the InvalidHL7Message() method or the FieldPrecisionMessage() method.
3. The front-end can then use the 'ErrorCode' to display a friendly message. i.e. In FileHandlerMessaging.tsx -> ErrorRow we call the ErrorCodeTranslation enum which maps the error codes to a friendly message.

Test Steps:
1. Log into RS as Admin
2. Click on the Validate feature and validate an hl7 file that contains invalid phone or a ValidationError
3. Notice the Error message is now friendly

## Changes
![Screenshot 2022-11-30 at 9 04 26 AM](https://user-images.githubusercontent.com/102491809/204861809-e6c36f20-f7f4-4ab7-ab45-27514f150a5c.png)
![Screenshot 2022-11-30 at 9 05 00 AM](https://user-images.githubusercontent.com/102491809/204861898-54240101-35a6-4de1-a168-20a291a4930f.png)


## Checklist

### Testing

## Linked Issues
- Fixes #7277
